code/gfx/d3d_layer.hxx

```diff
@@ -0,0 +1,47 @@
+// Enable discrite GPU
+//
+__declspec(dllexport) DWORD NvOptimusEnablement                = 0x00000001;
+__declspec(dllexport) int AmdPowerXpressRequestHighPerformance = 1;
+
+/**
+ * Typedefs of ID3D types to avoid interface versioning.
+ * D3D have various updates with increasing numbers as sufixes
+ * to help in maintaining code.
+ */
+
+using IDevice              = ::ID3D11Device;
+using IDeviceContext       = ::ID3D11DeviceContext;
+using ISwapChain           = ::IDXGISwapChain;
+using IRenderTargetView    = ::ID3D11RenderTargetView;
+using ITexture2D           = ::ID3D11Texture2D;
+using IRasterizerState     = ::ID3D11RasterizerState;
+using IComputeShader       = ::ID3D11ComputeShader;
+using IVertexShader        = ::ID3D11VertexShader;
+using IPixelShader         = ::ID3D11PixelShader;
+using IInputLayout         = ::ID3D11InputLayout;
+using IBuffer              = ::ID3D11Buffer;
+using IBlob                = ::ID3DBlob;
+using IUnorderedAccessView = ::ID3D11UnorderedAccessView;
+using IShaderResourceView  = ::ID3D11ShaderResourceView;
+using IQuery               = ::ID3D11Query;
```

Comment:
In DX11 you can always use newest interfaces available, since it's an older API. This way you can have more ways to accomplish tasks or to get another idea how to approach problem.